### PR TITLE
Docs - split template features by version

### LINKED ISSUE #147: Document what version introduces a specific feature
**Is your feature request related to a problem? Please describe.**
Document what version introduces a specific feature so that people know what to expect.

**Describe the solution you'd like**
- Only publish documentation for new features when a new version is released
- Provide indication in documentation what version introduces the feature

**Describe alternatives you've considered**
None

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #147

### CODE REVIEW

File: docs/features/web-api-template.md [hunk 1]
```diff
@@ -23,16 +23,31 @@ When installed, the template can be created with shortname: `arcus-webapi`:
 ## Features
```
Comment (tomkerkhove) [inline docs/features/web-api-template.md]: Actually I would go from:
- Feature 
	- Version 0.1
- Config
	- Version 0.1
- Feature 
	- Version 0.2
- Config
	- Version 0.2

To
- Feature 
	- Version 0.1
	- Version 0.2
- Config
	- Version 0.1
	- Version 0.2

Comment (stijnmoreels) [reply docs/features/web-api-template.md]: But then you have to look at two places to find the available functionality. Now the configuration and security and possibly other headers are sub headers of each version.

Comment (tomkerkhove) [reply docs/features/web-api-template.md]: I don't really agree so let's ask a 3rd person.

Currently have this @SamVanhoutte, what do you think?
image

Or should we use following approach?

Feature
- Version 0.1
- Version 0.2
Config
- Version 0.1
- Version 0.2

Comment (tomkerkhove) [reply docs/features/web-api-template.md:23]: Any feedback @SamVanhoutte ?

Comment (SamVanhoutte) [reply docs/features/web-api-template.md:23]: Not really following 100%, but I'd think that you typically are only interested in the version you are using, so why is there no proposal that has 
Version 0.2
- Feature
- Config
Version 0.1
- Feature 
- Config

Might have missed the point, so bear with me.  Also I believe the most recent version should be at the top?

Comment (tomkerkhove) [reply docs/features/web-api-template.md:23]: That's certainly an option but that assumes you'll duplicate the information from v0.1 in v0.2 then or not?

So what I actually suggest is to move this to the URL :)

`/features/templates/web-api` goes to latest version which is noted at top via badge. At the bottom you can find a list of old docs.
`/v0.1/features/templates/web-api` provides the old docs 

Then the docs are scoped per page and have capability to go back.

Thoughts?

Comment (SamVanhoutte) [reply docs/features/web-api-template.md:23]: That sounds logical, indeed.  
`As a user`, I just want to see the documentation of the version I'm using and most of the time I don't care about older versions.  Again: up to you two

Comment (tomkerkhove) [reply docs/features/web-api-template.md:23]: Let's use URLs then!

Comment (stijnmoreels) [reply docs/features/web-api-template.md:23]: This is actually a problem for every Arcus project.

Comment (tomkerkhove) [reply docs/features/web-api-template.md:23]: An opportunity you mean ;)

That's correct but nothing we can't fix. Let's start doing that here and if it's good we can roll it out to others

File: docs/features/web-api-template.md [hunk 2]
```diff
@@ -24,21 +24,7 @@ When installed, the template can be created with shortname: `arcus-webapi`:
```
Comment (tomkerkhove) [inline docs/features/web-api-template.md:24]: Let's drop a note here on the version we have under `Web API Project Template` or so with a link maybe?

Comment (stijnmoreels) [reply docs/features/web-api-template.md:24]: You mean specifying which version this is?

Comment (tomkerkhove) [reply docs/features/web-api-template.md:24]: List all the versions of the documentation such as:

> - Latest
> - [v0.1](/v0.1/features/web-api-template)

Comment (stijnmoreels) [reply docs/features/web-api-template.md:24]: I did on the bottom a list of the older versions? Isn't that enough?

File: docs/v0.1/features/web-api-template.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+> dotnet new --install Arcus.Templates.WebApi
+```
+
+When installed, the template can be created with shortname: `arcus-webapi`:
+
+```shell
+> dotnet new arcus-webapi --name Arcus.Demo.WebAPI
+```
+
+
```
Comment (tomkerkhove) [inline docs/v0.1/features/web-api-template.md:22]: Let's remove this line
```suggestion
```

File: docs/features/web-api-template.md (hunk 2 above)
Comment (tomkerkhove) [reply docs/features/web-api-template.md:24]: LGTM, missed that sorry!

**Pull request merged**